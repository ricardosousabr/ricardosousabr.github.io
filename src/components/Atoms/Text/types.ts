import { ReactNode } from "react";

export interface TextProps {
  children: ReactNode;
  color: string;
  fontSize: string;
}