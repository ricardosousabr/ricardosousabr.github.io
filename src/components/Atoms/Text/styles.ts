import styled from "styled-components";
import { TextProps } from "./types";

export const TextStyle = styled.p<TextProps>`
  color: ${(p) => p.color};
  font-size: ${(p) => p.fontSize};
`