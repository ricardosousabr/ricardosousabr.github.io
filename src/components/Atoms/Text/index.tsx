import { TextProps } from "./types"
import { TextStyle } from "./styles"

export default function Text({ children, color, fontSize }: TextProps) {
  return <TextStyle color={color} fontSize={fontSize}>{children}</TextStyle>
}
