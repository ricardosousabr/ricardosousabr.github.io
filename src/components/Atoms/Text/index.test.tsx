import renderer from "react-test-renderer";
import Text from "./index";

test("Link renders correctly", () => {
  const tree = renderer.create(<Text color="aqua" fontSize="16px">Click</Text>).toJSON();
  expect(tree).toMatchSnapshot();
});
