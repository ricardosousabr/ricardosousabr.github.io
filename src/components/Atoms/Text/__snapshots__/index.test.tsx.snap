// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<p
  className="sc-bdnyFh hgFkHH"
  color="aqua"
  fontSize="16px"
>
  Click
</p>
`;
